JsonExtensions/JsonReader.cs

```diff
@@ -121,7 +115,11 @@ public bool Read()
                 {
                     // there's space left in the buffer, try to fill it with new data
                     int todo = this.buffer.Length - this.dataLen;
-                    int done = this.Stream.Read(this.buffer, this.dataLen, todo);
+#if NET6_0_OR_GREATER
+                    int done = await this.Stream.ReadAsync(this.buffer.AsMemory(this.dataLen, todo));

```

Comment:
This is the important part. And the AsMemory call is to satisfy [CA1835](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1835)